lua/guard-collection/formatter.lua

```diff
@@ -137,8 +137,8 @@ M.pg_format = {
 }
 
 M.prettier = {
-  cmd = 'prettier',
-  args = { '--stdin-filepath' },
+  cmd = 'npx',
```

Comment:
what if one day latest prettier introduce breaking changes, like eslint doesn't support `.eslintrc`, global linters can't handle  all project that have their suitable version of linters

Reply:
in normal develop workflow, to avoid code style issue, there always is a commit hook to invoke prettier, eslint or tsc to auto fix fixable issues, but project owner can not assure every have installed linters locally globally, use wrong version of linters to fix code with not capable linter config may cause bugs

Reply:
Well, that would make this change a breaking change for some users, couldn't we do a workaround? It makes sense for people like me to have a global install or prettier from package manager or mason.

Reply:
![CleanShot 2024-05-29 at 16 35 05@2x](https://github.com/nvimdev/guard-collection/assets/59885141/2100e0f7-ec0d-4e83-a5a8-50c07b8d9c72)

https://docs.npmjs.com/cli/v7/commands/npx#description

> If any requested packages are not present in the local project dependencies, then they are installed to a folder in the npm cache, which is added to the PATH environment variable in the executed process.

npx will look for global installed version as fallback as you are not installed project level.

if use mason, maybe they need add mason package install path to `PATH`.

i was use mason, but after eslint v9 doesn't support `.eslintrc` file anymore, i have to reinstall eslint with specific version many times...

Reply:
Thanks for the clarification! I guess the question then is whether npx is likely available for everyone that has npm?

Reply:
![CleanShot 2024-05-29 at 16 49 06@2x](https://github.com/nvimdev/guard-collection/assets/59885141/69ea57b4-933c-44c2-9fa5-53eedce39185)

`npx` is alias of `npm exec`, `npm` is binding with nodejs, that means as long as you installed node, npm npx is also available

Reply:
> if use mason, maybe they need add mason package install path to PATH.

or add to guard init?